### ISSUE TITLE
Question: timeline with events for monitoring

### ISSUE THREAD

**Issue by bownie** [swimlane/ngx-charts#780]
I'd like to represent events on a timeline including stop/start times (therefore range bars), one-off and recurring events (annotations) and therefore have a new component or custom chart which is suitable for systems monitoring purposes.

I'd believe ngx-charts could be a great basis for this (as there are so many elements already included) but at a slight loss as to which demo or even custom chart to use as a starting point.  The fundamental data difference here is that I would want to represent many data points at the same time point on the single chart.  Can anyone point me in the right direction? Will of course be happy to share any results as a PR and I have some (but not unlimited) time to devote to this.

**Comment by bownie** [comment]
Ok I've done a bit of poking around and getting a bit excited about mixed angular and svg tags. Looks like this should be possible but best as a custom chart or a new component?  I'm thinking possibly the latter?

**Comment by marjan-georgiev** [comment]
Based on the description it looks like you're looking for a gantt chart? https://www.google.de/search?q=gantt+chart&tbm=isch

I think that is a pretty common chart type, and we would welcome any effort in building it as a first-class chart-type of the library.

If you're willing to start on that, we can discuss the API and data format in here.

**Comment by bownie** [comment]
Yes certainly a gannt chart basis along with the ability to indicate events which may be global or local to a particular series with the possibility of tooltips or labels on these events.

**Comment by cole21771** [comment]
I believe events would be useful on other charts as well, specifically for the line chart I could make use of that for sure. I would imagine that it might show a vertical bar with a small tag attached (vertically or horizontally, not sure which would be better) and in the case of live data it would move along with whatever point it is attached to as to indicate that something happened at that time.
